### ISSUE TITLE
Getting a "Query returned multiple rows; Cannot return scalar value." after doing "Take(1)"

### ISSUE THREAD

**Issue by kristofclaes** [ThatRendle/Simple.Data#61]
When I run this code when no previous photo should be found, I get an exception saying "Query returned multiple rows; Cannot return scalar value.". This is strange because it shouldn't return _any_ values and because I use ".Take(1)" to make sure only one row is returned.

```
string previousSlug = DB.Photos
                        .Query()
                        .Select(DB.Photos.Slug)
                        .Where(DB.Photos.Published == true && DB.Photos.DatePublished < photo.DatePublished.Value) // There is no published photo with a smaller DatePublished than the one I check against
                        .OrderByDatePublishedDescending()
                        .Take(1) // Should return only one record no matter what
                        .ToScalarOrDefault<string>();
```

**Comment by ThatRendle** [comment]
Do you have 0.6.7 installed? I thought this was fixed in that release. If not I'll take a look tomorrow.

**Comment by kristofclaes** [comment]
I'm not sure. I think I have 0.6.6, but I'll check it later today when I get home. I wasn't aware that 0.6.7 was available on Nuget.

**Comment by ThatRendle** [comment]
Yeah, I snuck 0.6.7 out while I was prepping my Nancy/Simple.Data demo for a talk, didn't have time to blog it. Give it a try, your issue was one of the things I fixed (and added tests for).

**Comment by kristofclaes** [comment]
I was indeed using 0.6.6. Apparently you released 0.6.7 exactly one day after I first encountered the problem :-) I just updated to 0.6.7 and the problem is fixed. Thanks for the help.
